lib/nfsbase.pm

```diff
@@ -0,0 +1,146 @@
+package nfsbase;
+use mmapi;
+use testapi;
+use strict;
+use warnings;
+use utils;
+use mm_network;
+use version_utils;
+
+
+our @ISA    = qw(Exporter);
+our @EXPORT = qw(server_configure_network try_nfsv2 prepare_exports yast_handle_firewall add_rw_share mount_export client_common_tests);
+
+sub server_configure_network {
+    my ($self) = @_;
+
+    # Configure static IP for client/server test
+    configure_default_gateway;
+    configure_static_ip('10.0.2.101/24');
+    configure_static_dns(get_host_resolv_conf());
+
+    if (is_sle('15+')) {
+        record_soft_failure 'boo#1130093 No firewalld service for nfs-kernel-server';
+        systemctl 'stop ' . $self->firewall;
+        systemctl 'disable ' . $self->firewall;
+    }
+}
+
+sub try_nfsv2 {
+    # Try that NFSv2 is disabled by default
+    systemctl 'restart nfsserver';
+    assert_script_run "cat /proc/fs/nfsd/versions | grep '\\-2'";
+    assert_script_run "sed -i 's/MOUNTD_OPTIONS=.*/MOUNTD_OPTIONS=\"-V2\"/g' /etc/sysconfig/nfs";
+    assert_script_run "sed -i 's/NFSD_OPTIONS=.*/NFSD_OPTIONS=\"-V2\"/g' /etc/sysconfig/nfs";
+    systemctl 'restart nfsserver';
+    assert_script_run "cat /proc/fs/nfsd/versions | grep '+2'";
+
+    # Disable NFSv2 again
+    assert_script_run "sed -i 's/MOUNTD_OPTIONS=.*/MOUNTD_OPTIONS=\"\"/g' /etc/sysconfig/nfs";
+    assert_script_run "sed -i 's/NFSD_OPTIONS=.*/NFSD_OPTIONS=\"\"/g' /etc/sysconfig/nfs";
+    systemctl 'stop nfsserver';
+}
+
+sub prepare_exports {
+    # Create a directory and place a test file in it
+    assert_script_run 'mkdir /srv/nfs && echo success > /srv/nfs/file.txt';
```

Comment:
you could assign the /srv/nfs and /srv/dir to variables to avoid hardcoding usage